containers/netty-http/src/main/java/org/glassfish/jersey/netty/httpserver/JerseyServerHandler.java

```diff
@@ -173,8 +181,28 @@ public void removeProperty(String name) {
                         properties.remove(name);
                     }
                 }, resourceConfig);
+    }
+
+    private String getPath(HttpRequest req) {
+        String s = req.uri();
+        if (baseUriHasBasePath()) {
+            s = s.startsWith("/") ? s.substring(1) : s;
+            if (!baseUri.toString().endsWith("/")) {
```

Comment:
It is unclear what this condition does, but it is not needed for either of your tests

Reply:
for case when base is: http://localhost:9994/base
and resource is: /base/resource/
the request URI will be: http://localhost:9994/baseresource/

I suppose that is not what's expected, however tests are passing OK. 
I suppose that's not which is expected

Reply:
Yes, that's the case. It's a bit weird,  but the tests now reflect that.

Reply:
so, you are saying, that having "http://localhost:9994" + "/baseresource/" is expected behavior?

Reply:
This is mixing the baseUri and path segments into a single segment, which is not what has been intended. It would bring confusion to the users.

Reply:
Ok, that makes sense. I was thinking it would be better not to change the code-server, so I made the changes there. 
After some digging I rediscovered the problem root. 
When the baseUri doesn't end with a delimiter, a call to the ContainerRequest#encodedRelativePath will retrieve a relative path starting with a delimiter, which will then result in a double starting delimiter when called from MatchResultInitializerRouter#apply. That will result in a non matching path when resolving in  RoutingStage.
I tried two solutions:
The first is adding the delimiter, on the baseUri, when creating the ContainerRequest (in JerseyServerHandler#createContainerRequest). The other is adding it on ContainerRequest#encodedRelativePath.
I think those are the simplest options, and both pass the tests.
If you guys can give me your opinion, i can make another commit with the changes. 
Thanks